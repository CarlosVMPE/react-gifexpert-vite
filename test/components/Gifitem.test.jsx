import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test in <GifItem />", () => {
  const title = "Dragon Ball";
  const url = "https://dbz.com/image.jpg";
  test("should be match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Should be show the img with the URL and ALT", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    // one option
    /* expect(screen.getByRole('img').src).toBe(url);
    expect(screen.getByRole('img').alt).toBe(title); */
    // two option
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should be show the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })
});
