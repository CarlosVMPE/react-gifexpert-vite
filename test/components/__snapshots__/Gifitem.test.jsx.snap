// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem /> should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://dbz.com/image.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
